runtime/vm/OutOfLineINL_jdk_internal_foreign_abi_ProgrammableInvoker.cpp

```diff
@@ -0,0 +1,162 @@
+/*******************************************************************************
+ * Copyright (c) 2021, 2021 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include "VMHelpers.hpp"
+#include "BytecodeAction.hpp"
+#include "UnsafeAPI.hpp"
+#include "j9vmnls.h"
+#include "omrlinkedlist.h"
+#include "OutOfLineINL.hpp"
+#include "FFITypeHelpers.hpp"
+#include "AtomicSupport.hpp"
+
+extern "C" {
+
+#if JAVA_SPEC_VERSION >= 16
+/* jdk.internal.foreign.abi.ProgrammableInvoker: private static synchronized native void resolveRequiredFields(); */
+VM_BytecodeAction
+OutOfLineINL_jdk_internal_foreign_abi_ProgrammableInvoker_resolveRequiredFields(J9VMThread *currentThread, J9Method *method)
```

Comment:
I dont fully understand why this is needed, I was under the impression that this would be done in `initializeKnownClasses`

Reply:
Nope, this can't be accomplished by `initializeKnownClasses()` in the initailization stage because the `jclconstantpool` is initialized too early and can only depend on `java.base` which `ProgrammableInvoker` doesn't belong to, as discussed with @pshipton.

Another alternative is move `ProgrammableInvoker` to `java.base` with a different package & name. But it will end up with tons of unnecessary changes on OpenJDK side in replacing `ProgrammableInvoker` with the new package/name given `ProgrammableInvoker`  is used everywhere on all platforms.  To avoid this case, we chose to resolve these offsets in the static initializer.